From ef142c2d628312e32d9940a1e0ded0145505510f Mon Sep 17 00:00:00 2001
From: Tomasz Zawadzki <tomasz.zawadzki@intel.com>
Date: Mon, 17 Aug 2020 05:03:43 -0400
Subject: [PATCH 3/5] pci/linux: free the device if no kernel driver configured

Patch (c79a1c67) made pci_scan_one() return without adding
it to the list of devices on the bus.
Since RTE_KDRV_NONE is no longer used, free the dev
before returning.

Signed-off-by: Tomasz Zawadzki <tomasz.zawadzki@intel.com>
Change-Id: I3bf0d4173f1dbfa7be94aacd382350b6acde907e
Reviewed-on: https://review.spdk.io/gerrit/c/spdk/dpdk/+/3817 (spdk-20.08)
Reviewed-on: https://review.spdk.io/gerrit/c/spdk/dpdk/+/5339 (spdk-20.11)
Reviewed-on: https://review.spdk.io/gerrit/c/spdk/dpdk/+/6429 (spdk-21.02)
Signed-off-by: Krzysztof Karas <krzysztof.karas@intel.com>
Reviewed-on: https://review.spdk.io/gerrit/c/spdk/dpdk/+/8137
Reviewed-by: Aleksey Marchuk <alexeymar@mellanox.com>
Reviewed-by: Ben Walker <benjamin.walker@intel.com>
---
 drivers/bus/pci/linux/pci.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/bus/pci/linux/pci.c b/drivers/bus/pci/linux/pci.c
index 0dc99e9..909502e 100644
--- a/drivers/bus/pci/linux/pci.c
+++ b/drivers/bus/pci/linux/pci.c
@@ -332,6 +332,7 @@
 			dev->kdrv = RTE_PCI_KDRV_UNKNOWN;
 	} else {
 		dev->kdrv = RTE_PCI_KDRV_NONE;
+		free(dev);
 		return 0;
 	}
 	/* device is valid, add in list (sorted) */
-- 
1.8.3.1

